server/storage/mvcc/index.go

```diff
@@ -257,6 +257,9 @@ func (ti *treeIndex) Equal(bi index) bool {
 	equal := true
 
 	ti.tree.Ascend(func(item btree.Item) bool {
+		if _, ok := b.tree.Get(item).(*keyIndex); !ok {
+			return false
+		}
 		aki := item.(*keyIndex)
 		bki := b.tree.Get(item).(*keyIndex)
```

Comment:
```suggestion
		if bki, ok := b.tree.Get(item).(*keyIndex); !ok {
			return false
		}
		aki := item.(*keyIndex)
```
Should we also check `aki`?